### DISCUSSION TITLE
Can I Override a Built-In Function?

### DISCUSSION THREAD

**Discussion by ghost** [codeceptjs/CodeceptJS#4309] [Q&A] [answered]
Currently, I am using CodeceptJS to automate API testing. The original built-in function, such as sendGetRequest, works well, but it doesn't generate any logs for export to Allure. 
Therefore, I would like to override this built-in function to generate logs for export to Allure.
Is there any way ?

**Comment by kobenguyent** [comment, marked as answer]
![Screenshot 2024-04-18 at 09 41 03](https://github.com/codeceptjs/CodeceptJS/assets/7845001/aef6c817-4d89-4946-b4ba-994c0de3cbc4)
perhaps you could try define the function with same name in custom steps file https://codecept.io/pageobjects/#actor then it would override the built-in function.

```
Before(async () => {
	newUser = await I.have('user', null, null);
	id = newUser.id;
});

Scenario.only('Verify creating new user', () => {
	I.sendGetRequest('hello', '123');
...

    I have "user", , 
    › Creating user {"name":"Eulalia Schoen","job":"leader"}
    › [Request] {"method":"POST","url":"/api/users","data":{"name":"Eulalia Schoen","job":"leader"},"baseURL":"https://reqres.in","headers":{"Content-Type":"application/json","Accept":"application/json"}}
    › [Response] {"name":"Eulalia Schoen","job":"leader","id":"350","createdAt":"2024-04-18T07:40:12.606Z"}
    › [Created] Id: 350
    --- ENDED "before each" hook: Before for "Verify creating new user @Jaja" ---
overridden
```
